stable_baselines3/common/noise.py

```diff
@@ -84,3 +85,55 @@ def reset(self) -> None:
 
     def __repr__(self) -> str:
         return f'OrnsteinUhlenbeckActionNoise(mu={self._mu}, sigma={self._sigma})'
+
+
+class VectorizedActionNoise(ActionNoise):
+    """
+    A Vectorized action noise for parallel environments.
+
+    :param base_noise: ActionNoise The noise generator to use
+    :param n_envs: (int) The number of parallel environments
+    """
+
+    def __init__(self, base_noise: ActionNoise, n_envs: int):
+        super().__init__()
+        self.n_envs: int = n_envs
+        self.noises = [copy.deepcopy(base_noise) for _ in range(n_envs)]
+
+    def reset(self, indices: Optional[Iterable[int]] = None) -> None:
+        """
+        Reset all the noise processes, or those listed in indices
+
+        :param indices: Optional[Iterable[int]] The indices to reset. Default: None.
+            If the parameter is None, then all processes are reset to their initial position.
+        """
+        if indices is None:
+            indices = range(len(self.noises))
+
+        for index in indices:
+            self.noises[index].reset()
+
+    def __repr__(self) -> str:
+        base_repr = repr(self.noises[0]) if len(self.noises) else ""
+        return f"VecNoise(BaseNoise={base_repr}), n_envs={len(self.noises)})"
+
+    def __call__(self) -> np.ndarray:
+        """
+        Generate and stack the action noise from each noise object
+        if n_envs == 0, returns a 0x0 matrix.
+        """
+        if len(self.noises):
+            noise = np.stack([noise() for noise in self.noises])
+        else:
+            noise = np.zeros((0, 0))
+        return noise
+
+    @property
+    def noises(self) -> List[ActionNoise]:
+        return self._noises
+
+    @noises.setter
+    def noises(self, noises: List[ActionNoise]) -> None:
+        self._noises = noises
```

Comment:
you should check the length too, no?

Reply:
I also need to ensure it is not None.

Reply:
I meant `assert len(noises) == n_envs`

Reply:
Now that I think of it, if I put a setter, then it's assumed to be public, so it needs sanity checks, yes?

Reply:
yes, that's what I meant